vyper/semantics/analysis/local.py

```diff
@@ -758,15 +758,8 @@ def _analyse_range_call(node: vy_ast.Call) -> list[VyperType]:
     """
     validate_call_args(node, (1, 2), kwargs=["bound"])
     kwargs = {s.arg: s.value for s in node.keywords or []}
-    arg0 = node.args[0].get_folded_value if node.args[0].has_folded_value else node.args[0]
-    start, end = (
-        (vy_ast.Int(value=0), arg0)
-        if len(node.args) == 1
-        else (
-            arg0,
-            node.args[1].get_folded_value() if node.args[1].has_folded_value else node.args[1],
-        )
-    )
+    start, end = (vy_ast.Int(value=0), node.args[0]) if len(node.args) == 1 else node.args
+    start, end = [i.get_folded_value() if i.has_folded_value else i for i in (start, end)]
```

Comment:
hm maybe we want to defer this until we are actually doing the exception handling a few lines down, need to think on this